proxysql_galera_checker

```diff
@@ -38,7 +38,61 @@ check_permission(){
     exit 1
   fi
 }
+# upgrade scheduler from old layout to new layout
+upgrade_scheduler(){
+  if [[ -f /etc/proxysql-admin.cnf ]]; then
+    source /etc/proxysql-admin.cnf
+  else
+    echo "Assert! proxysql-admin configuration file :/etc/proxysql-admin.cnf does not exists, Terminating!"
+    exit 1
+  fi
 
+  execute()
+  {
+  local query="$1"
+  printf "%s\n" \
+      "[client]" \
+      "user='${PROXYSQL_USERNAME}'" \
+      "password='${PROXYSQL_PASSWORD}'" \
+      "host=${PROXYSQL_HOSTNAME}" \
+      "port=${PROXYSQL_PORT}"  \
+      | timeout 2 mysql --defaults-file=/dev/stdin --protocol=tcp -Nse "${query}"
+  if [ "$?" == "124" ]; then
+    echo "PROXYSQL-ADMIN TIMEOUT ERROR: Connection terminated due to timeout."
+  fi
+  }
+
+  SCHED=$(execute "SELECT * FROM scheduler")
+  for i in "${SCHED[@]}"; do
+    ID=$(echo "$i" | awk '{print $1}')
+    S_WRITE_HG=$(echo "$i" | awk '{print $5}')
+    S_READ_HG=$(echo "$i" | awk '{print $6}')
+    S_NUMBER_OF_WRITES=$(echo "$i" | awk '{print $7}')
+    S_LOG=$(echo "$i" | awk '{print $9}')
+    S_CLUSTER_NAME=$(echo "$i" | awk '{print $10}')
+    if [[ -z $S_CLUSTER_NAME ]]; then
+      if [[ -f ${PROXYSQL_DATADIR}/mode ]] ; then
+        S_MODE=$(cat ${PROXYSQL_DATADIR}/mode)
+      fi
+    else
+      if [[ -f ${PROXYSQL_DATADIR}/${S_CLUSTER_NAME}_mode ]] ; then
+        S_MODE=$(cat ${PROXYSQL_DATADIR}/${S_CLUSTER_NAME}_mode)
+      fi
+    fi
+    if [ -f $HOST_PRIORITY_FILE ];then
```

Comment:
Should this also check the --priority parameter?

Reply:
@kennt-percona `--priority` is a new parameter, if that is set it means they already have the new layout.